.circleci/config.yml

```diff
@@ -0,0 +1,105 @@
+version: 2.1
+executors:
+  default-executor:
+    docker:
+      - image: "cimg/node:12.22.11-browsers"
+    working_directory: ~/project
+    resource_class: medium
+
+commands:
+  restore_test_cache:
+    steps:
+      - restore_cache:
+          keys:
+            - v1-deps-{{ checksum "package-lock.json" }}
+            - v1-deps-
+  save_test_cache:
+    steps:
+      - save_cache:
+          key: v1-deps-{{ checksum "package-lock.json" }}
+          paths:
+            - node_modules
+            - ~/.cache/ms-playwright
+
+  setup:
+    steps:
+      - run:
+          name: Install dependencies
+          command: |
+            npm --production=false install
+            npm --production=false update
+            npx playwright install
+
+  test:
+    steps:
+      - run:
+          name: Test
+          command: |
+            npm run test
+  setup_deploy:
+    steps:
+      - run:
+          name: Setup Deploy
+          command: |
+            echo "export NPM_TAG=latest" >> $BASH_ENV
+            echo "export NODE_ENV=production" >> $BASH_ENV
+            echo "export RELEASE_TIMESTAMP=$(date +'%Y%m%d%H%M%S')" >> $BASH_ENV
+            npm run build
+            sudo chmod +x .circleci/before_deploy.sh
+            .circleci/before_deploy.sh
```

Comment:
Since this is the only place it's used, it seems like it would make sense to just move the `before_deploy.sh` contents into this script. Unless there's a reason not to do that...?

If it does need to be in a different file, I'd recommend doing a `chmod +x before_deploy.sh` and then `git commit` that, so you don't need the `chmod` step here.

Reply:
Moved script into main file